src/reducer/FilterReducer.jsx

```diff
@@ -0,0 +1,80 @@
+const FilterInitialState = {
+  category: {
+    All: false,
+    Laptop: false,
+    Gaming_Laptops: false,
+    CPU: false,
+    Phones: false,
+  },
+  rating: "",
+  sortBy: "",
+  price: 100000,
+};
+
+const FilterReducer = (state, {type, payload}) => {
+  switch (type) {
+    case "ALL":
+      return {
+        ...state,
+        category: {
+          ...state["category"],
+          All: !state.category.All,
+          Gaming_Laptops: false,
+          Laptop: false,
+          CPU: false,
+          Phones: false,
+        },
+      };
```

Comment:
these things are not required. Maybe unknowingly you have added them. You can remove that

Reply:
I am using them in my filterReducer